### ISSUE TITLE
Uint8Array assignable to ArrayBuffer despite having subtle differences leading to browser TypeErrors

### ISSUE THREAD

**Issue by peaBerberian** [microsoft/TypeScript#42534]
# Bug Report

### 🔎 Search Terms

ArrayBuffer
Uint8Array

### 🕗 Version & Regression Information

At least in:
  - typescript 4.1.3
  - nightly

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.2.0-dev.20210128#code/G4QwTgBCELwQdgUwO4QKoEt4BcAcBBMMEATwAoBtARgBoIAmOgZgF0BKAbgCgBjAe3gBnPgBtEAOhF8A5mSSoAIiGwgAahhRkQbTkA)

### 💻 Code

```ts
var a = new Uint8Array([1, 2, 3]);
new DataView(a);
```

Also, to be more explicit about the root issue in TypeScript:
```ts
function createDataView(buffer : ArrayBuffer) : DataView {
  return new DataView(buffer);
}

var a = new Uint8Array([1, 2, 3]);
createDataView(a);
```
### 🙁 Actual behavior

This is kind of continuation of the since closed due to inactivity https://github.com/microsoft/TypeScript/issues/31311 issue, but with other arguments.

The sample codes presented here go through TypeScript's checks despite both leading to a browser TypeError because `DataView`'s constructor expects an `ArrayBuffer`, not an `Uint8Array` (on FireFox, the error is: `TypeError: DataView: expected ArrayBuffer, got Uint8Array`).

From the linked issue, it seems that TypeScript considers an Uint8Array to be a valid ArrayBuffer because they are structurally the same. But there are subtle differences like this one.

I understand the point of view that TypeScript is not supposed to be perfectly sound, but I think that it would be nice if TypeScript caught those types of mistakes.

We had an issue related to this and I think that we wouldn't have it with plain JavaScript, because we would have been more careful with types! In the end, not always thinking about these types of mistakes is a major reason for switching to TS.

### 🙂 Expected behavior

I would expect that an ArrayBuffer and a Uint8Array to not  be considered the same thing.

**Comment by MartinJohns** [comment]
As mentioned in #31311, this is a duplicate of #202 (or it would require #202 due to the nature of TypeScripts type system). Wesley even showed a [workaround](https://github.com/microsoft/TypeScript/issues/31311#issuecomment-490690695).

**Comment by peaBerberian** [comment]
OK thanks.

My bad, I have skipped over that one comment strangely.
I saw the last comment in the issue write about a "duplicate" but I didn't see which one it was speaking of.

The work-around should work and I saw it but this issue was more about integrating that logic into TS instead (as I didn't see the duplicates, I took that comment as a refusal to include it in TS).

I'm closing that issue now.
